v6-t-test-py/partial.py

```diff
@@ -40,28 +43,40 @@ def partial(df: pd.DataFrame, column_name: str) -> dict:
             f"{MINIMUM_NUMBER_OF_RECORDS}."
         )
 
-    if not ptypes.is_numeric_dtype(df[column_name]):
-        error("Column must be numeric.")
+    if not columns:
+        columns = df.select_dtypes(include=["int", "float"]).columns.tolist()
```

Comment:
There are more numeric types in the `pandas` universe than the simple `int` and `float` (although maybe this are container types too, not sure). I checked the documentation of pandas and found that you can use numeric to get all numeric types:
```suggestion
        columns = df.select_dtypes(include=["number"]).columns.tolist()
```

See the first bullet point in: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.select_dtypes.html